packages/jest-preset-mc-app/fail-on-console.js

```diff
@@ -0,0 +1,123 @@
+// This module is based on the package `jest-fail-on-console` but with some custom adjustments.
+const util = require('util');
+const colors = require('colors/safe');
+
+const defaultErrorMessage = (methodName, bold) =>
+  `Expected test not to call ${bold(`console.${methodName}()`)}.\n\n` +
+  `If the ${methodName} is expected, test for it explicitly by mocking it out using ${bold(
+    'jest.spyOn'
+  )}(console, '${methodName}').mockImplementation() and test that the warning occurs.`;
+
+const failOnConsole = ({
+  errorMessage = defaultErrorMessage,
+  shouldFailOnAssert = false,
+  shouldFailOnDebug = false,
+  shouldFailOnLog = false,
+  shouldFailOnInfo = false,
+  shouldFailOnWarn = true,
+  shouldFailOnError = true,
+  skipTest,
+  silenceMessage,
+  logButNotThrowMessage,
+} = {}) => {
+  const flushUnexpectedConsoleCalls = (
+    methodName,
+    unexpectedConsoleCallStacks
+  ) => {
+    if (unexpectedConsoleCallStacks.length > 0) {
+      const messages = unexpectedConsoleCallStacks.map(([stack, message]) => {
+        const stackLines = stack.split('\n');
+        return (
+          `${colors.red(message)}\n` +
+          `${stackLines
+            .map((line, index) => {
+              if (index === stackLines.length - 1) {
+                return colors.white(line);
+              }
+              return colors.gray(line);
+            })
+            .join('\n')}`
+        );
+      });
+
+      const message = errorMessage(methodName, colors.bold);
+
+      throw new Error(`${message}\n\n${messages.join('\n\n')}`);
+    }
+  };
+
+  const patchConsoleMethod = (methodName) => {
+    let originalMethod = console[methodName];
+
+    const unexpectedConsoleCallStacks = [];
+
+    const captureMessage = (format, ...args) => {
+      const message = util.format(format, ...args);
+
+      if (silenceMessage && silenceMessage(message, methodName)) {
+        return;
+      }
+
+      if (logButNotThrowMessage && logButNotThrowMessage(message, methodName)) {
+        originalMethod(format, ...args);
+        return;
+      }
```

Comment:
This is the only functionality (that we had in our implementation) "missing" from the original plugin. Therefore, I opted to simply copy over the implementation and adjust it accordingly.

Reply:
How much is this custom functionality is important? Any chance we can drop it and simply use the original plugin as you did in `test-data`?

Reply:
This replaces this functionality:

https://github.com/commercetools/merchant-center-application-kit/blob/237472109fbb27898b3b02e7b0c2a269a7ac95ca/packages/jest-preset-mc-app/setup-tests.js#L71-L73

If we want we can remove it, but it would probably be a breaking change. We could include the change once we support Jest v29 maybe.